src/App.tsx

```diff
@@ -8,53 +8,104 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getPosts, getUser } from './api/api';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUser()
+      .then(setUsers)
+      .catch(() => setIsError(true));
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    if (activeUser) {
+      setIsLoading(true);
+      getPosts({ id: activeUser.id })
+        .then(setPosts)
+        .catch(() => setIsError(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [activeUser]);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const isPostsList =
+    activeUser && !isError && posts.length > 0 ? posts : false;
```

Comment:
The `isPostsList` logic is incorrect. It should return a boolean value instead of the posts array or false. Consider changing this logic to `!!(activeUser && !isError && posts.length > 0)` to return a boolean.